handshake-api/index.js

```diff
@@ -0,0 +1,75 @@
+const bodyparser = require("body-parser");
+const express = require("express");
+const { NodeClient } = require("hs-client");
+
+const host = process.env.HOST || "localhost";
+const port = Number(process.env.PORT) || 3100;
+
+const hsdNetworkType = process.env.HSD_NETWORK || "regtest";
+const hsdHost = process.env.HSD_HOST || "localhost";
+const hsdPort = Number(process.env.HSD_PORT) || 12037;
+const hsdApiKey = process.env.HSD_API_KEY || "foo";
+
+const clientOptions = {
+  network: hsdNetworkType,
+  host: hsdHost,
+  port: hsdPort,
+  apiKey: hsdApiKey,
+};
+const client = new NodeClient(clientOptions);
+
+const resolveDomain = async (name) => {
+  const response = await client.execute("getnameresource", [name]);
+
+  if (!response) throw new Error("API not responding");
+
+  console.log(`${name} => ${JSON.stringify(response.records)}`);
```

Comment:
Potential nullpointer on empty response. We are accessing them directly without checking their existence on line 34 again, so we might just as well verify them here and throw a "no records found" exception.

Reply:
It's not a problem here because we check `response` on line 24.